### PR TITLE
Query as Params

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**
The requests library supports query strings to be sent for a request in the params arument See https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls. This is a safer and neater approach than embedding them as raw strings in the url.

### CODE REVIEW

File: modelon/impact/client/sal/workspace.py [hunk 1]
```diff
@@ -226,24 +209,20 @@ def get_published_workspaces(
...
-            query["workspaceName"] = name
-        if has_data:
-            query["hasData"] = str(has_data)
+        query = {
+            "workspaceName": name,
+            "hasData": has_data,
+            "ownerUsername": owner_username,
+            "type": type,
+            "groupName": group_name,
+        }
```
Comment (iakovn) [inline modelon/impact/client/sal/workspace.py:218]: Why did you skip all the conditions? It seems unnecessary to pass, e.g.,  `workspaceName=None` all the way to the server?

Comment (abhilash-kumar-nair) [reply modelon/impact/client/sal/workspace.py:218]: I felt this makes the python client api consistent, if the defaults in server are updated for some reason? The server would anyway parse the query strings and set defaults if they are not set here.  I agree for workspaceName probably it doesnt make sense to send in.

Comment (iakovn) [reply modelon/impact/client/sal/workspace.py:218]: I agree with this argument for boolean fields (like  `has_data`) and eventually pagination control (first/max) but I feel it is a bit weird for optional strings like workspaceName, ownerUsername and groupName. Are those getting serialized like `workspaceName=null` when passed to requests?

Comment (abhilash-kumar-nair) [reply modelon/impact/client/sal/workspace.py:218]: Just read the docs, seems like requests would not append it to the URL if its **NONE** its seems as per docs - https://docs.python-requests.org/en/latest/user/quickstart/#passing-parameters-in-urls

For empty string they would just be added like https://blahblah.com/api?workspaceName=&ownerUsername=

Comment (iakovn) [reply modelon/impact/client/sal/workspace.py:218]: Thanks for double-checking! Then I'm fine with the update.

**Comment by iakovn** [comment]
LGTM! 👍

**Review by iakovn — APPROVED** [review]

**Pull request merged**